pkg/build/controller/config_controller.go

```diff
@@ -33,11 +37,32 @@ func IsFatal(err error) bool {
 
 type BuildConfigController struct {
 	BuildConfigInstantiator buildclient.BuildConfigInstantiator
+
+	KubeClient kclient.Interface
+	Client     osclient.Interface
 }
 
 func (c *BuildConfigController) HandleBuildConfig(bc *buildapi.BuildConfig) error {
 	glog.V(4).Infof("Handling BuildConfig %s/%s", bc.Namespace, bc.Name)
 
+	if strategy := bc.Spec.Strategy.JenkinsPipelineStrategy; strategy != nil {
+		glog.V(4).Infof("Detected Jenkins pipeline strategy in %s/%s build configuration", bc.Namespace, bc.Name)
+		if _, err := c.KubeClient.Services(bc.Namespace).Get(buildapi.JenkinsDefaultServiceName); err == nil {
```

Comment:
@jimmidyson what I observed is that when I forgot to create the jenkins template, but I created this BC, nothing really happened. But as soon as I created the template it deployed automatically... I guess this will be the same case.